snuba/clickhouse/native.py

```diff
@@ -111,13 +114,16 @@ def execute(
         try:
             conn = self.pool.get(block=True)
 
-            attempts_remaining = 3
+            attempts_remaining = 3 + (
+                1 if USE_FALLBACK_HOST_IN_NATIVE_CONNECTION_POOL else 0
+            )
+            use_fallback_host = False
             while attempts_remaining > 0:
                 attempts_remaining -= 1
                 # Lazily create connection instances
                 if conn is None:
                     self.__gauge.increment()
-                    conn = self._create_conn()
+                    conn = self._create_conn(use_fallback_host)
```

Comment:
This is going to be a problem. If you assign the fallback connection to `conn`, this will be put back into the main connections pool at line 202. This means that following queries would end up picking this connection from that pool even at the first attempt to run a query. 

I would suggest keeping this conneciton around separately from the main conneciton pool after you have created it. It can be a separate (smaller) queue. So that when you decide you want to use a fallback connection, you pick it from that queue instead of recreating it new. and during normal operation the main queue will still rely on envoy